### PR TITLE
[WIP][DO NOT MERGE] CI Travis: add  shell code style shellcheck

### PR DESCRIPTION
**Pull request by xiulipan**
Add Travis CI to check all shell scripts with shellcheck
Add build status in README.md

### CODE REVIEW

**Comment by xiulipan** [comment]
@marc-hb @Bin-QA @aiChaoSONG I think we may should have a clean up for the existing warning and errors.

File: README.md [hunk 1]
```diff
@@ -10,6 +10,10 @@
     - [tools](#tools)
   - [Folder description](#folder-description)
   - [Tools list description](#tools-list-description)
+
+## Status
+[![Build Status](https://travis-ci.org/thesofproject/sof-test.svg?branch=master)](https://travis-ci.org/thesofproject/sof-test)
+
```
Comment (Bin-QA) [inline README.md]: can you also update the `Table of contents` at top?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  - sudo apt-get update
+  - sudo apt-get -y install shellcheck
+
+stages:
+  - code-check
+
+jobs:
+  include:
+    - stage: code-check
+      name: "bash shellcheck"
```
Comment (marc-hb) [inline .travis.yml]: shellcheck is not just for bash (I think it may even report some bashisms)

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      script: shellcheck */*.sh
```
Comment (marc-hb) [inline .travis.yml]: Anyone who fixes 1 line and sees 100s of warnings reported on code that has not changed will immediately start ignore this check and will never look back.

At the very least this check should run only on the files changed in the PR. Or even better: https://github.com/AtakamaLLC/lint-diffs (not tested)

Comment (xiulipan) [reply .travis.yml]: @marc-hb so you think this will only test with patches or the whole files?
I can refine this.

**Comment by xiulipan** [comment]
@marc-hb Thanks for the suggestion, I will try to do some update for this scripts to have check for patches.

**Comment by marc-hb** [comment]
> https://github.com/AtakamaLLC/lint-diffs (not tested)

If this doesn't work or takes too long to get working, then as a second best run it on the entire `.sh` script but ONLY for the files that have been modified in the PR. For examples have a look at the `--diff-filter` in the battle-hardened https://github.com/zephyrproject-rtos/ci-tools/blob/master/scripts/check_compliance.py

**Comment by marc-hb** [comment]
@xiulipan can you rebase on top of this commit: https://github.com/marc-hb/sof-test/blob/shellcheck-gitrange/tools/shellcheck-gitrange.bash

and then use in .travis.yml, something like `./tools/shellcheck-gitrange.bash ${TRAVIS_COMMIT_RANGE}`?

**Comment by marc-hb** [comment]
With @xiulipan 's permission I'm taking this task.

**Comment by marc-hb** [comment]
Completed in #183, thanks in advance for reviewing.

**Pull request closed without merging**